syntax = "proto3";
import "default_msg/default.proto";

service DbConnectorService {
    rpc GetBalance(UserId) returns (BalanceInfo);
    rpc GetUserInfo(UserId) returns (UserInfo);
    rpc Login(LoginData) returns (UserInfo);
}

message UserId {
    string user_id = 1;
}

message LoginData {
    string login = 1;
    string password = 2;
}

message BalanceInfo {
    float balance = 1;
    string currency = 2;
}

message UserInfo {
    string id = 1;
    string first_name = 2;
    string last_name = 3;
    string phone_number = 4;
    string address = 5;
    string login = 6;
}
